docs/index.rst

```diff
@@ -127,6 +127,7 @@ nature, you get a fast DataFrame at 1MB and 1TB+.
 
    using_modin
    out_of_core
+   modin_xgboost
```

Comment:
Maybe we worth using `modin_on_ray_xgboost`?

Reply:
I think `modin_xgboost` is better because this file contains documentation for general interfaces/behavior for Modin XGBoost. In case adding `Dask` interface, section for "Single node / Cluster setup" will be expanded only.